src/nvim/terminal.c

```diff
@@ -907,6 +928,16 @@ static void on_refresh(Event event)
       }
       continue;
     }
+    FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+      if (wp->w_buffer == term->buf) {
+        if ( wp->w_botline > wp->w_buffer->b_ml.ml_line_count ) {
+          // if last line is visible, follow the terminal
+          wp->following_terminal = true;
+        } else {
+          wp->following_terminal = false;
+        }
```

Comment:
If I open a new terminal buffer and run a command, the scroll back should be empty. If "scrollback should always have at least one line", doesn't that mean it won't scroll?

BTW: I realized that when you press enter, the cursor does move down one, so that means even on a new terminal window, the trick of moving up one after starting the command will still work.

But on the surface, sounds like a good plan. :+1: